client/my-sites/domains/index.js

```diff
@@ -247,7 +247,7 @@ export default function() {
 			domainsController.redirectIfNoSite( '/domains/add' ),
 			domainsController.redirectToAddMappingIfVipSite(),
 			jetPackWarning,
-			domainsController.domainSearch,
+			domainsController.redirectToDomainSearchWithSuggestionAsQuery,
 			makeLayout,
 			clientRender
```

Comment:
If the last middleware always redirects, we don't need `makeLayout` and `clientRender` here anymore. They just handle the final rendering (which now will happen in some other route).